### ISSUE TITLE
Strange behavior when array elements moved one level up and the last one deleted

### ISSUE THREAD

**Issue by miyconst** [Juicy/dom-bind-notifier#3]
If an array item deleted by moving the next items up one level and removing the last element, then something strange happens and next time such operation removes latest element without moving items up.

Here is a jsfiddle to demonstrate the issue: https://jsfiddle.net/bk546a1n/.
Press the `Remove row #1` button multiply times and the issue will be there.

This issue discovered this: https://github.com/Starcounter/Starcounter/issues/2920.

cc @tomalec

**Comment by tomalec** [comment]
It seems that our workaround (`translatePathForPolymerNotifyPath`) for inconsistent Polymer API (https://github.com/Polymer/polymer/issues/2248) does not work in that case.
Or one of many bugs in Polymer:
 - https://github.com/Polymer/polymer/issues/2470

**Comment by tomalec** [comment]
It seems to be a conflict with Polymer's dirty checking, as we are notifying add of an element, which is already there. So we are adding another reference to an element that already exists 
```js
       (var i = 1) {
            var item = template.model.items[i + 1];
            template.model.items[i] = { name: item.name, index: item.index };
        }
```
`template.model.items[i + 1]` might be equal for Polymer to `template.model.items[i]`


(BTW, it's not moving, it's re-adding elements, as https://github.com/Starcounter/Starcounter/issues/2920#issuecomment-144381678)

**Comment by tomalec** [comment]
Fixed together with #4
